### PR TITLE
Added a struct to hide the output of a std::cin

### PR DESCRIPTION
**Pull request by Lambourl**

### CODE REVIEW

File: cli_widgets/stdin_hide_guard.hpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      DWORD mode;
+      GetConsoleMode(hStdin, &mode);
+      mode |= ENABLE_ECHO_INPUT;
+      SetConsoleMode(hStdin, mode);
+#else
+      struct termios tty;
+      tcgetattr(STDIN_FILENO, &tty);
+      tty.c_lflag |= ECHO;
+      (void)tcsetattr(STDIN_FILENO, TCSANOW, &tty);
+#endif
```
Comment (daminetreg) [inline cli_widgets/stdin_hide_guard.hpp]: Well the implementation before with a central function that you could pass true and false  was cleanier. I mean you could call the function inside the RAII wrapper.

Comment (daminetreg) [reply cli_widgets/stdin_hide_guard.hpp]: It's just a matter of oring/xoring the ECHO flag, then the code can be divided by half in size.

File: cli_widgets/stdin_hide_guard.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#include <boost/predef.h>
+#if BOOST_OS_WINDOWS
+#include <windows.h>
+#else
+#include <termios.h>
+#include <unistd.h>
+#endif
+
+namespace cli_widgets{
+  struct stdin_hide_guard {
```
Comment (daminetreg) [inline cli_widgets/stdin_hide_guard.hpp]: Perhaps adding a brief to document  the purpose  would be cool

**Review by daminetreg — CHANGES_REQUESTED** [review]

**Comment by Lambourl** [comment]
I thought that for the RAII, I had to have a destroyer which returns to the initial state

**Comment by Lambourl** [comment]
I think I understood the structure is only a wrapper

**Comment by daminetreg** [comment]
Could you add : 
- [x] an example usage in an examples/ folder
- [x]  a documentation comments on the class 
- [x] pull master to have the CI run through

?

File: .nxxm/deps [hunk 3]
```diff
@@ -1,4 +1,5 @@
 { 
   "boostorg/predef" : { "@" : "master" },
   "ikalnitsky/termcolor" : { "@" : "master" }
+  ,"platform" : [ "Boost::thread", "Boost::system", "Boost::filesystem", "Boost::regex" ]
```
Comment (daminetreg) [inline .nxxm/deps]: We don't need them here.

**Review by daminetreg — APPROVED** [review]

**Comment by daminetreg** [comment]
I believe hidden_input_guard is finally the best name for this

**Review by daminetreg — APPROVED** [review]
Thanks ! 👍

**Pull request merged**